### PR TITLE
Schemaless tap

### PR DESCRIPTION
**Pull request by JingLin0**

### CODE REVIEW

File: tap_capterra/clicks.py [hunk 1]
```diff
@@ -31,6 +33,15 @@ def get_clicks(start_date, end_date, api_key):
...
 
 
+@backoff.on_exception(
+    backoff.expo,
+    (
+        requests.exceptions.RequestException,
+        requests.exceptions.HTTPError,
+        ratelimit.exception.RateLimitException,
+    ),
+    max_tries=5,
```
Comment (paddie) [inline tap_capterra/clicks.py:42]: max_tries <3

File: tap_capterra/clicks.py [hunk 2]
```diff
@@ -31,6 +33,15 @@ def get_clicks(start_date, end_date, api_key):
                 break
 
 
+@backoff.on_exception(
+    backoff.expo,
+    (
+        requests.exceptions.RequestException,
+        requests.exceptions.HTTPError,
+        ratelimit.exception.RateLimitException,
```
Comment (paddie) [inline tap_capterra/clicks.py:40]: I actually think that `requests.exceptions.RequestException` encompasses `requests.exceptions.HttpError`.. Or is it the other way around? My point is, we might only need one of these?

File: tap_capterra/clicks.py [hunk 3]
```diff
@@ -44,20 +44,12 @@ def get_clicks(start_date, end_date, api_key):
...
-    except HTTPError as http_err:
-        logger.error(f"HTTP error occurred: {http_err}")
-    except Exception as err:
-        logger.error(f"Other error occurred: {err}")
+    response = SESSION.get(
+        "https://public-api.capterra.com/v1/clicks",
+        headers={"Accept": "application/json", "Authorization": f"Bearer {api_key}"},
+        params=params,
+    )
+    response.raise_for_status()
```
Comment (paddie) [inline tap_capterra/clicks.py:51]: nice, using the backoff directly. Well done.

File: tap_capterra/clicks.py [hunk 4]
```diff
@@ -21,7 +20,7 @@ def get_clicks(start_date, end_date, api_key):
         "total": len(list(rrule(DAILY, dtstart=start_date, until=end_date))),
         "unit": "day",
     }
-    for date in tqdm(rrule(DAILY, dtstart=start_date, until=end_date), **kwargs):
+    for date in rrule(DAILY, dtstart=start_date, until=end_date):
```
Comment (paddie) [inline tap_capterra/clicks.py:23]: `dtstart`?

Comment (JingLin0) [reply tap_capterra/clicks.py:23]: Ah, `rrule` is a library function. I can't change its arg names.

Comment (paddie) [reply tap_capterra/clicks.py:23]: Ahh, fine then :D

**Pull request merged**